runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -96,6 +96,7 @@ public void gc(long trimMark) {
      */
     public List<SMREntry> current() {
         ILogData data = streamView.current();
+
         if (data == null
                 || data.getType() != DataType.DATA
                 || !(data.getPayload(runtime) instanceof ISMRConsumable)) {
```

Comment:
Sorry for going back and forth on this, but can we reintroduce the NO-OP SMR ? I think its better to be more concise and handle holes explicitly in the object layer instead of just using an empty list as a signal.